.circleci/config.yml

```diff
@@ -0,0 +1,32 @@
+version: 2
+
+references:
+  cache_key: &cache_key
+    key: jars-{{ checksum "build.gradle" }}-{{ checksum  "app/build.gradle" }}-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}
+
+jobs:
+  build:
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx3200m
```

Comment:
Way too much memory allocated here. Circle has a 4G limit, but here you're giving each JVM 3.2GB. Gradle by default will create workers = CPU count, so that's 12GB. When R8 becomes default, that will be done out of process too.

I'd suggest limiting your workers and max heap. See my [Tivi](https://github.com/chrisbanes/tivi/blob/master/.circleci/config.yml#L8) for an example. Also disable the daemon as that uses RAM and is unnecessary for CI.

Reply:
Good to know.  Tweaked environment settings.